public/docs/dart/latest/guide/forms.jade

```diff
@@ -0,0 +1,662 @@
+
+figure.image-display
+  img(src="/resources/images/devguide/forms/hf-1.png" alt="Clean Form")
+
+:marked
+  Here at the *Hero Employment Agency* we use this form to maintain personal information about the
+  heroes in our stable. Every hero needs a job. It's our company mission to match the right hero with the right crisis!
+
+  Two of the three fields on this form are required. Required fields have a green bar on the left to make them easy to spot.
+
+  If we delete the hero name, the form displays a validation error in an attention-grabbing style:
+
+figure.image-display
+  img(src="/resources/images/devguide/forms/hf-2.png" alt="Invalid, Name Required")
+
+:marked
+  Note that the submit button is disabled, and the "required" bar to the left of the input control changed from green to red.
+
+.l-sub-section
+  p We'll customize the colors and location of the "required" bar with standard CSS.
+
+:marked
+  We'll build this form in small steps:
+
+  1. Create the `Hero` model class.
+  1. Create the component that controls the form.
+  1. Create a template with the initial form layout.
+  1. Add the **ng-model** directive to each form input control.
+  1. Add the **ng-control** directive to each form input control.
+  1. Add custom CSS to provide visual feedback.
+  1. Show and hide validation error messages.
+  1. Handle form submission with **ng-submit**.
+  1. Change the form's display after submission.
+
+:marked
+  ## Setup
+  Create a new project folder (`angular2_forms`) and create 3 files:
+  `pubspec.yaml`, `web/index.html`, and `web/main.dart`.
+  (These files should be familiar from the
+  [QuickStart](../quickstart.html).) Put these contents in the files:
+
++makeTabs('forms/dart/angular2_forms-1-SETUP/pubspec.yaml, forms/dart/angular2_forms-1-SETUP/web/index.html, forms/dart/angular2_forms-1-SETUP/web/main.dart', null, 'pubspec.yaml, web/index.html, web/main.dart')
+
+:marked
+  So that the code can run,
+  let's create a stub for the `<hero-form>` component.
+
+  Create a new directory called `lib`.
+  In it, put a file called `hero_form_component.dart`
+  with the following code:
+
++makeExample('forms/dart/angular2_forms-1-SETUP/lib/hero_form_component.dart', null, 'lib/hero_form_component.dart')
+
+:marked
+  The app should now run, but it won't do anything interesting.
+  Let's add some data.
+
+
+  ## Create the Hero model class
+
+  As users enter form data, we'll capture their changes and update an instance of a model.
+  We can't lay out the form until we know what the model looks like.
+
+  A model can be as simple as a "property bag" that holds facts about a thing of application importance.
+  That describes well our `Hero` class with its three required fields (`id`, `name`, `power`)
+  and one optional field (`alterEgo`).
+
+  In the `lib` directory, add a file called `hero.dart`
+  with the following code:
+
++makeExample('forms/dart/angular2_forms-2-COMPONENT/lib/hero.dart', 'all', 'lib/hero.dart')
```

Comment:
Good idea adding the file name to the example title bar

I notice you do this elsewhere when showing a fragment of a file. I find this confusing ... because it isn't the whole file.

One way to be clear would be to add the suffix " (excerpt)" at the end as in "web/index.html (excerpt)"